src/main/java/edu/kit/compiler/semantic/NamespaceMapper.java

```diff
@@ -0,0 +1,42 @@
+package edu.kit.compiler.semantic;
+
+import edu.kit.compiler.data.ast_nodes.ClassNode;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class NamespaceMapper {
+
+    private final Map<ClassNode, ClassNamespace> symbolTableMap = new HashMap<>();
```

Comment:
Could we also map from identifier (i.e. `int`) to `ClassNamespace` and include a `ClassNode` attribute in `ClassNamespace`? That way you could easily map from a `DataType` to the corresponding class and access its attributes/fields.

Reply:
Yes, i remember that we discussed this earlier for the symbol table as well. I think its good to just stick to one solution, therefore we should only reference our symbols and classes with their respective id not with the data type itself.

Reply:
It's also relevant when you have, for example, a `Definition` of a variable with a `DataType` and want to access the list of methods available on that variable.

Reply:
Why I asked to include `ClassNode` is, if you have an "active" namespace (i.e. a variable `currentNamespace : ClassNamespace` and encounter a `this` pointer you have to find the class data type, which is (for now) not stored in the namespace.